.ci/test.sh

```diff
@@ -64,8 +64,9 @@ if [[ "$TASK" == "cpp-tests" ]]; then
     exit 0
 fi
 
-# including python=version[build=*cpython] to ensure that conda doesn't fall back to pypy
-CONDA_PYTHON_REQUIREMENT="python=${PYTHON_VERSION}[build=*cpython]"
+# including python=version=[build=*_cp*] to ensure that conda prefers CPython and doesn't fall back to
+# other implementations like pypy
+CONDA_PYTHON_REQUIREMENT="python=${PYTHON_VERSION}[build=*_cp*]"
```

Comment:
Found that changes on this line were the reason for the failures I saw in https://github.com/microsoft/LightGBM/pull/6668#issuecomment-2572125384.

`python=3.9=*_cp*` and `python=3.9[build=*_cp*]` are handled differently by `conda`. I reported that at https://github.com/conda/conda-libmamba-solver/issues/604, but we should just continue with the `[build` form here as it does work as expected.